CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+    end
+
+  result =
+    if some_cond
+      calc_something
+    else
+      calc_something_else
+    end
+  ```
+
+* Use empty lines between method definitions and also to break up a method into logical paragraphs internally.
+
+  ```Ruby
+  def some_method
+    data = initialize(options)
+
+    data.manipulate!
+
+    data.result
+  end
+
+  def some_method
+    result
+  end
+  ```
+
+* Avoid comma after the last parameter in a method call, especially when the parameters are not on separate lines.
+
+  ```Ruby
+  # bad - easier to move/add/remove parameters, but still not preferred
+  some_method(
+               size,
+               count,
+               color,
+             )
+
+  # bad
+  some_method(size, count, color, )
+
+  # good
+  some_method(size, count, color)
+  ```
+
+* Avoid line continuation `\` where not required. In practice, avoid using line continuations for anything but string concatenation.
+
+  ```Ruby
+  # bad
+  result = 1 - \
+           2
+
+  # good (but still ugly as hell)
+  result = 1 \
+           - 2
+
+  long_string = 'First part of the long string' \
+                ' and second part of the long string'
+  ```
+
+* Adopt a consistent multi-line method chaining style. There are two popular styles in the Ruby community, both of which are considered good - leading `.` (Option A) and trailing `.` (Option B).
+
+  * **(Option A)** When continuing a chained method invocation on
+    another line keep the `.` on the second line.
+
+    ```Ruby
+    # bad - need to consult first line to understand second line
+    one.two.three.
+      four
+
+    # good - it's immediately clear what's going on the second line
+    one.two.three
+      .four
+    ```
+
+  * **(Option B)** When continuing a chained method invocation on another line,
+    include the `.` on the first line to indicate that the
+    expression continues.
+
+    ```Ruby
+    # bad - need to read ahead to the second line to know that the chain continues
+    one.two.three
+      .four
+
+    # good - it's immediately clear that the expression continues beyond the first line
+    one.two.three.
+      four
+    ```
+
+  A discussion on the merits of both alternative styles can be found
+  [here](https://github.com/bbatsov/ruby-style-guide/pull/176).
+
+* Align the parameters of a method call if they span more than one line. When aligning parameters is not appropriate due to line-length constraints, single indent for the lines after the first is also acceptable.
+
+  ```Ruby
+  # starting point (line is too long)
+  def send_mail(source)
+    Mailer.deliver(to: 'bob@example.com', from: 'us@example.com', subject: 'Important message', body: source.text)
+  end
+
+  # bad (double indent)
+  def send_mail(source)
+    Mailer.deliver(
+        to: 'bob@example.com',
+        from: 'us@example.com',
+        subject: 'Important message',
+        body: source.text)
+  end
+
+  # good
+  def send_mail(source)
+    Mailer.deliver(to: 'bob@example.com',
```

Comment:
I personally just put the args on the next line to keep the whitespace down, that way you only need to indent by 2